src/parser/hays.py

```diff
@@ -11,325 +10,303 @@
-                    if row[0] == "Bondsman":
-                        break
-                state_rows = state_rows[::-1]
-                defendant_rows = defendant_rows[::-1]
-                # Get the position of the item containing gender
-                gender_position = 0
-                for i, row in enumerate(defendant_rows[0]):
-                    if row[:5] == "Male " or row[:7] == "Female ":
-                        gender_position = i
-                    if row[:5] == ["DOB: "]:
-                        dob_position = i
-                # Handle "Also Known as" which will mess with positions
-                if gender_position > 2:
-                    full_name = " ".join(defendant_rows[0][1:gender_position])
-                    defendant_rows[0] = (
-                        [defendant_rows[0][0]]
-                        + [full_name]
-                        + defendant_rows[0][gender_position:]
-                    )
-                if gender_position == 0:
-                    defendant_rows[0].insert(2, "Unavailable Unavailable")
+            return top_charge
+        except Exception as e:
+            logger.info(f"Error getting top charge: {e}")
+            return {
+                "charge name": "Unknown",
+                "charge level": "Unknown"
+            }
 
-                bondsman_rows = bondsman_rows[::-1]
-                if bondsman_rows[0][0] != "Bondsman":
-                    bondsman_rows = []
+    def get_case_metadata(self, county: str, case_number: str, case_soup: BeautifulSoup, logger) -> Dict[str, str]:
+        try:
+            logger.info(f"Getting case metadata for {county} case {case_number}")
+            return {
+                "code": case_soup.select('div[class="ssCaseDetailCaseNbr"] > span')[0].text,
+                "odyssey id": case_number,
+                "county": county
+            }  
+        except Exception as e:
+            logger.info(f"Error getting case metadata: {e}")
+            return {
+                "code": "Unknown",
+                "odyssey id": case_number,
+                "county": county
+            }
+        
+    def get_case_details(self, table: BeautifulSoup, logger) -> Dict[str, str]:
+        try:
+            table_values = table.select("b")
+            logger.info(f"Getting case details")
+            return {
+                "name": table_values[0].text,
+                "case type": table_values[1].text,
+                "date filed": table_values[2].text,
+                "location": table_values[3].text
+            }
+        except Exception as e:
+            logger.info(f"Error getting case details: {e}")
+            return {
+                "name": "Unknown",
+                "case type": "Unknown",
+                "date filed": "Unknown",
+                "location": "Unknown"
+            }   
+    
+    def parse_defendant_rows(self, defendant_rows: List[List[str]], logger) -> Dict[str, str]:
```

Comment:
Just a comment. This is the hays-specific parser, and I imagine that this data will be different from one county to the next. I imagine this would be very much county-specific, but as we get into different counties, we'll probably think of creating a helpers functions that works with other functions that don't vary much from county to county. TBD.